eventespresso/barista
# Modify Ticket Status When Datetime Status Changes

With the new Datetime status changer, it is not obvious why a ticket may not be available for purchase if its related datetime makes it unavailable (like a cancelled datetime).

This PR:

- modifies a ticket's status if it's related datetimes make it unavailable for purchase
- displays a brief notice above the ticket explaining what has happened

### testing

- add/edit an event that has multiple dates and tickets. I used 2 dates and 2 tickets so that i could test combinations of 1:1 vs 1:many relations.
- click the `*` button beside a datetime's displayed status and choose a new option
- [ ] confirm that related ticket statuses change logically. Plz note that statuses do not translate between entities exactly. For example, changing a datetime's status to `POSTPONED` will change a related ticket's status to `PENDING` (tickets do not have a `POSTPONED` status). Plz also note that tickets with more than one datetime will NOT change unless ALL of the datetimes result in the ticket being unavailable. Ticket statuses in these cases may be inconsistent (for example, a ticket with two related datetimes where one is `POSTPONED` and the other is `SOLD_OUT` could end up having its status changed to `PENDING` or `SOLD_OUT` depending on the order things happen. This doesn't really matter so plz don't over think it)
